src/osd/ECTransaction.cc

```diff
@@ -193,9 +193,6 @@ void ECTransaction::generate_transactions(
       xattr_rollback[ECUtil::get_hinfo_key()] = old_hinfo;
 
       if (op.is_none() && op.truncate && op.truncate->first == 0) {
-	ceph_assert(op.truncate->first == 0);
-	ceph_assert(op.truncate->first ==
-	       op.truncate->second);
 	ceph_assert(entry);
 	ceph_assert(obc);
```

Comment:
A dozen lines below there is the `if (obc)` which seems to be always taken. Maybe it's worth a follow-up?

Reply:
@rzarzynski yeah, I considered the same thing, but ultimately decided to leave it for a future PR.